### PR TITLE
docs: clarify use of browser.close

### LINKED ISSUE #15163: [BUG] `browser.close` doesn't close contexts properly
**Context:**
## System:
 - OS: Windows 10 10.0.19044
 - Memory: 1.41 GB / 15.78 GB
## Binaries:
 - Node: 18.0.0 - C:\Program Files\nodejs\node.EXE
 - npm: 8.7.0 - C:\Program Files\nodejs\npm.CMD
## Languages:
 - Bash: 5.0.17 - C:\WINDOWS\system32\bash.EXE
## npmPackages:
 - playwright: ^1.23.0 => 1.23.0

**Code Snippet**

```javascript
const { chromium } = require('playwright')

;(async () => {
    const browser = await chromium.launch()
    const context = await browser.newContext({
        recordHar: { path: 'asdf.har' },
    })
    const page = await context.newPage()
    await page.goto('https://google.com/')
    await browser.close()
})()
```

**Describe the bug**

when closing the browser with `browser.close()` instead of `context.close()`, it does not close the context properly. in this case, no HAR file is created.

i would expect `browser.close()` to also call `context.close()` on all of its contexts before closing the browser

### PR DESCRIPTION
**Pull request by rwoll**
Resolves #15163.

### CODE REVIEW

File: docs/src/api/class-browser.md [hunk 1]
```diff
@@ -92,6 +92,10 @@ were opened).
 In case this browser is connected to, clears all created contexts belonging to this browser and disconnects from the
 browser server.
 
+:::note
+This is similar to force quitting the browser. Therefore, you should call [`method: BrowserContext.close`] on any [`BrowserContext`]'s you explicitly created with [`method: Browser.newContext`] **before** calling [`method: BrowserContext.close`].
```
Comment (KotlinIsland) [inline docs/src/api/class-browser.md:96]:
```suggestion
This is similar to force quitting the browser. Therefore, you should call [`method: BrowserContext.close`] on any [`BrowserContext`]'s you explicitly created with [`method: Browser.newContext`] **before** calling [`method: Browser.close`].
```

File: packages/playwright-core/types/types.d.ts [hunk 2]
```diff
@@ -13224,6 +13224,12 @@ export interface Browser extends EventEmitter {
    * In case this browser is connected to, clears all created contexts belonging to this browser and disconnects from the
    * browser server.
    *
+   * > NOTE: This is similar to force quitting the browser. Therefore, you should call
+   * [browserContext.close()](https://playwright.dev/docs/api/class-browsercontext#browser-context-close) on any
+   * [`BrowserContext`]'s you explicitly created with
+   * [browser.newContext([options])](https://playwright.dev/docs/api/class-browser#browser-new-context) **before** calling
+   * [browserContext.close()](https://playwright.dev/docs/api/class-browsercontext#browser-context-close).
```
Comment (KotlinIsland) [inline packages/playwright-core/types/types.d.ts:13231]:
```suggestion
   * [browser.close()](https://playwright.dev/docs/api/class-browsercontext#browser-context-close).
```

**Review by KotlinIsland — CHANGES_REQUESTED** [review]

File: docs/src/api/class-browser.md [hunk 3]
```diff
@@ -156,6 +160,11 @@ Returns the newly created browser session.
 
 Creates a new browser context. It won't share cookies/cache with other browser contexts.
 
+:::note
+If directly using this method to create [`BrowserContext`]s, it is best practice to explicilty close the returned context via [`method: BrowserContext.close`] when your code is done with the [`BrowserContext`],
+and before calling [`method: Browser.close`]. This will ensure the `context` is closed gracefully and any artifacts—like HAR's—are fully flushed and saved.
```
Comment (yury-s) [inline docs/src/api/class-browser.md:165]: nit `like HAR's and videos`?

**Review by yury-s — APPROVED** [review]

File: docs/src/api/class-browser.md (hunk 1 above)
Comment (rwoll) [reply docs/src/api/class-browser.md]: Updated—thank you!

**Comment by rwoll** [comment]
Let me update the examples below this note so they follow the note!

**Pull request merged**